src/coord/src/catalog/migrate.rs

```diff
@@ -257,6 +258,62 @@ pub const CONTENT_MIGRATIONS: &[fn(&mut Catalog) -> Result<(), anyhow::Error>] =
         tx.commit()?;
         Ok(())
     },
+    // Insert default value for confluent_wire_format
+    //
+    // Before this change we expected all Avro-encoded messages over Kafka to
+    // come prepended with a magic byte and a schema registry ID, and so some
+    // folks modified their input stream to include those magic bytes. To
+    // preserve the old behavior for existing catalogs in case we change the
+    // default to a possibly more reasonable one in the future, we ensure that
+    // the current default is encoded in the on-disk catalog.
+    //
+    // Introduced for v0.7.1
+    |catalog: &mut Catalog| {
+        let mut storage = catalog.storage();
+        let items = storage.load_items()?;
+        let tx = storage.transaction()?;
+
+        for (id, name, def) in items {
+            let SerializedCatalogItem::V1 {
+                create_sql,
+                eval_env,
+            } = serde_json::from_slice(&def)?;
+
+            let mut stmt = sql::parse::parse(&create_sql)?.into_element();
+
+            // the match arm is long enough that this is easier to understand
+            #[allow(clippy::single_match)]
+            match stmt {
+                Statement::CreateSource(CreateSourceStatement {
+                    format:
+                        Some(Format::Avro(AvroSchema::Schema {
+                            ref mut with_options,
+                            ..
+                        })),
+                    ..
+                }) => {
+                    if with_options.is_empty() {
```

Comment:
It seems odd to me that if you've e.g. specified `client_id` or `security_protocol` that we wouldn't need this new default. idk much about these integrations, though, so quite possibly something I misunderstand.

Reply:
ah, this is specifically the `schema` with options, kafka sources have several divergent/distinct with option pragmas. I'll add a comment. I think this is possibly a larger problem, but I'm not sure how to solve it.

![image](https://user-images.githubusercontent.com/277161/109338811-c5607e80-7834-11eb-99d7-2a853c45a9ae.png)

Reply:
It's probably more clear to the reader if we check whether the options contain `confluent_wire_format` specifically, not just whether they're empty.

Reply:
ty, Brandon; I didn't even realize we had options in multiple places now so obviously didn't read carefully enough.